app/mailers/user_email_mailer.rb

```diff
@@ -32,4 +25,37 @@ def reminder_email(user, response, time_period)
     @time_period = time_period
     mail(to: user.email, subject: "#{user.first_name}, your check-in was saved.")
   end
+
+  private
+
+  def set_links(user, time_period)
+    general_link = URL.merge({ time_period_id: TimePeriod.current, not_working: false, user_id: user.id })
+    @link_for_own_word = general_link.merge({ last_step: 'emotion-entry' })
+    @link_for_was_not  = general_link.merge({ last_step: 'results' })
+    @link_for_emotion  = general_link.merge({ emotion_id: nil, last_step: 'meme-selection' })
+    @view_complete_by  = time_period.due_date.strftime('%b %d').to_s
+  end
+
+  def set_table_of_emotions
+    @table = generate_table(emotions_for_table)
+  end
+
+  def emotions_for_table
+    Emotion.where(public: true)
+           .positive.sample(NUMBER) +
+      Emotion.where(public: true)
+             .neutral.sample(NUMBER) +
+      Emotion.where(public: true)
+             .negative.sample(NUMBER)
+  end
+
+  def generate_table(emotions)
+    table = []
+    emotions.each_slice(6) { |sliced_emotions| table << sliced_emotions }
+    table.transpose.flatten
+  end
+
+  def send_invite_email(user)
+    mail(to: user.email, subject: "Hey #{user.first_name}, how has work been?")
+  end
 end
```

Comment:
@LyubovP Oops) This file no need to change because in here no have influence on the nonpublic emotions, by the way this file changed in the CI-338. Therefore if you want make refactoring need to wait when CI-388 merge with master.

Reply:
@VitaliiShevchenko Sorry, this bug is not related to the task CI-338 [Sub-Epic] Update Emotion Selection Email/Web pages, CI-464 is a bug related to the branch master currently in production. Regarding refactoring, I want to clarify that it is a recommendation from Codeclimate and the decision to refactor is not within my plans(.

Reply:
Ok, no problem